### DISCUSSION TITLE
Graphics artifacts when changing window size on X11

### DISCUSSION THREAD

**Discussion by ZoltanLajosKis** [ghostty-org/ghostty#3710] [Feature Requests, Ideas] [answered]
When resizing the window it looks like the old size is first stretched to the new size and content is only re-rendered afterwards.

Captured on Ubuntu 22.04.5 with X11, using Ghostty 1.0.0 built with zig-0.13 (-release=fast)

https://github.com/user-attachments/assets/1b56a906-55d1-420f-96e0-f21a5c19431d

Possibly related to https://github.com/ghostty-org/ghostty/issues/3682, but not sure.

**Comment by rhodes-b** [comment, marked as answer]
This is expected behavior, as the termio thread and graphics thread are separate this is on purpose for performance reasons, 
and I believe it was fixed by
https://github.com/ghostty-org/ghostty/pull/7155

I think this discussion explains it much better so I will mark this as a duplicate / answered
https://github.com/ghostty-org/ghostty/discussions/6016
